scrapy/core/downloader/webclient.py

```diff
@@ -119,15 +123,16 @@ def __init__(self, request, timeout=180):
             # just in case a broken http/1.1 decides to keep connection alive
             self.headers.setdefault("Connection", "close")
         # Content-Length must be specified in POST method even with no body
-        elif self.method == 'POST':
+        elif self.method == b'POST':
             self.headers['Content-Length'] = 0
 
     def _build_response(self, body, request):
         request.meta['download_latency'] = self.headers_time-self.start_time
         status = int(self.status)
         headers = Headers(self.response_headers)
-        respcls = responsetypes.from_args(headers=headers, url=self.url)
-        return respcls(url=self.url, status=status, headers=headers, body=body)
+        respcls = responsetypes.from_args(headers=headers, url=self._url)
+        body = to_bytes(body)
```

Comment:
Added a test in 673df5e4161337136154eeccfc4e327d053dbd12 - original encoding and headers are preserved.